internal/sim/corpus.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sim
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// # Simulator Corpus
```

Comment:
Reading the change description, I thought that a "corpus" was going to represent a series of `Simulators`, i.e., a collection of simulation runs with different setting and randomizations. Instead, a corpus is a set of failed runs, which is less intuitive IMO.

Maybe a better name for a collection of failed runs is something that includes failure in the name. "RIPs" is a playful example of this. If you're less worried about the name confusion, fine to leave as-is too.

Reply:
What about corpses instead of corpus :stuck_out_tongue_closed_eyes:, just kidding. 

Good point. The go fuzzing library uses the term corpus, but they do differ from us in that you can manually add non-failing inputs to the corpus. What do you think about renaming corpus to *graveyard* and corpus entry to *graveyard entry*? Also happy with something a little more bland like *failing corpus* or *counterexample corpus* or something like that.

Reply:
"Graveyard" sounds great to me!

Reply:
Renamed!